components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -318,4 +321,54 @@ void PageContentExtractor::GetSearchSummarizerKey(
   std::move(callback).Run(element.GetAttribute("content").Utf8());
 }
 
+void PageContentExtractor::ValidateOpenLeoButtonNonce(
+    const std::string& nonce,
+    mojom::PageContentExtractor::ValidateOpenLeoButtonNonceCallback callback) {
+  if (nonce.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto element = render_frame()->GetWebFrame()->GetDocument().GetElementById(
+      "continue-with-leo");
+  if (element.IsNull() || !element.HasHTMLTagName("a")) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  GURL url(element.GetAttribute("href").Utf8());
+  if (!IsOpenLeoButtonFromBraveSearchURL(url) ||
```

Comment:
optional as there would be some StrCat, but this `IsOpenLeoButtonFromBraveSearchURL` could be made into part of the QuerySelector using a startsWith attribute selector:

```
[href^="https://search.brave.com/leo#"]
```

additionally, this check doesn't test whether we have the `#$nonce` part of the url, just whether the `/leo` part of the path is present - probably isn't too important though

Reply:
`url.ref_piece() != nonce` in the same if covers it already.